src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -310,13 +310,22 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     std::map<std::size_t, std::string> forced_sub_devices{};
     const std::string fsd_opt = m_cfg.get<::intel_npu::NPUW_SUBMODEL_DEVICE>();
     forced_sub_devices = ::intel_npu ::OptionParser<std::map<std::size_t, std::string>>::parse(fsd_opt);
+
+    // Exclude optimized out subgraphs from compilation target beforehand - otherwise we might get head and repeated block in the same chunk
+    std::vector<std::size_t> idx_subgraph_to_compile;
+    for (std::size_t i = 0u; i < orderedSubgraphs.size(); i++) {
+        if (!orderedSubgraphs[i]._optimized_out) {
```

Comment:
If `orderedSubgraphs[i]._optimized_out || (m_compiled_submodels[i].replaced_by && m_compiled_submodels[i].replaced_by != i)` then we should skip